tests/test_main.cpp

```diff
@@ -378,6 +383,28 @@ int main( int argc, const char *argv[] )
         DebugLog( D_INFO, DC_ALL ) << "Default randomness seeded to: " << rng_get_first_seed();
     }
 
+    // Tests not requiring the global game initialized are tagged with [nogame]
+    {
+        using namespace Catch;
+        Config const &config = session.config();
+        std::vector<TestCase> const &tcs = filterTests(
+                                               getAllTestCasesSorted( config ),
+                                               config.testSpec(),
+                                               config
+                                           );
+        for( TestCase const &tc : tcs ) {
+            for( std::string const &tag : tc.getTestCaseInfo().tags ) {
+                // NOLINTNEXTLINE(cata-tests-must-restore-global-state)
+                needs_game = true;
+                if( tag == "nogame" ) {
+                    // NOLINTNEXTLINE(cata-tests-must-restore-global-state)
+                    needs_game = false;
```

Comment:
This will make `needs_game` false if a non-`[nogame]` test is followed by a `[nogame]` test.